src/workspace.coffee

```diff
@@ -612,16 +612,18 @@ class Workspace extends Model
 
   # Increase the editor font size by 1px.
   increaseFontSize: ->
+    @originalFontSize ?= atom.config.get("editor.fontSize")
     atom.config.set("editor.fontSize", atom.config.get("editor.fontSize") + 1)
 
   # Decrease the editor font size by 1px.
   decreaseFontSize: ->
+    @originalFontSize ?= atom.config.get("editor.fontSize")
     fontSize = atom.config.get("editor.fontSize")
     atom.config.set("editor.fontSize", fontSize - 1) if fontSize > 1
 
-  # Restore to a default editor font size.
+  # Restore to the window's original editor font size.
   resetFontSize: ->
-    atom.config.unset("editor.fontSize")
+    atom.config.set("editor.fontSize", @originalFontSize)
```

Comment:
Are you sure this will work the same as `atom.config.unset` in the case where `@originalFontSize` is `undefined`? Let's say my normal font size is 28. If I *do not* increase or decrease the font size and accidentally hit restore, will it set me to the Atom default font size? Or will it do something freaky?

There should also be a spec for this, that reset with no increase or decrease first should leave the font size where it is.